syz-manager/manager.go

```diff
@@ -654,9 +655,45 @@ func (mgr *Manager) loadCorpus() []fuzzer.Candidate {
 		candidates = append(candidates, item)
 	}
 	log.Logf(0, "%-24v: %v (%v seeds)", "corpus", len(candidates), seeds)
+	// Let's favorize smaller programs, otherwise the poorly minimized ones may overshadow the rest.
+	sort.SliceStable(candidates, func(i, j int) bool {
+		return len(candidates[i].Prog.Calls) < len(candidates[j].Prog.Calls)
+	})
+	reminimizeSubset(candidates)
 	return candidates
 }
 
+// reminimizeSubset clears the fuzzer.ProgMinimized flag of a small subset of seeds.
+// The ultimate objective is to gradually clean up the poorly minimized corpus programs.
+// reminimizeSubset assumes that candidates are sorted in the order of ascending len(Prog.Calls).
+func reminimizeSubset(candidates []fuzzer.Candidate) {
+	if len(candidates) < 100 {
+		return
+	}
+	// Only consider the top 10% of the largest programs.
+	threshold := len(candidates[len(candidates)*9/10].Prog.Calls)
+	var resetIndices []int
+	for i, info := range candidates {
+		if info.Flags&fuzzer.ProgMinimized == 0 {
+			continue
+		}
+		if len(info.Prog.Calls) >= threshold {
+			resetIndices = append(resetIndices, i)
+		}
+	}
+	// Reset ProgMinimized for up to 1% of the seed programs.
+	reset := min(50, max(1, len(candidates)/100))
+	if reset > len(resetIndices) {
+		reset = len(resetIndices)
+	}
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for _, i := range rnd.Perm(len(resetIndices))[:reset] {
+		idx := resetIndices[i]
+		candidates[idx].Flags &= ^fuzzer.ProgMinimized
+	}
+	log.Logf(1, "%d corpus seeds will be re-minimized", reset)
```

Comment:
I would fold this into corpus info logged in loadCorpus.
If you return this number from this function, it should be easy.